test/unit/plugins/provisioners/shell/provisioner_test.rb

```diff
@@ -43,6 +43,37 @@
     end
   end
 
+  context "with a script that was set to freeze the string" do
+    TEST_CONSTANT_VARIABLE = <<-TEST_CONSTANT_VARIABLE.freeze
+      echo test
+    TEST_CONSTANT_VARIABLE
+
+    let(:script) { TEST_CONSTANT_VARIABLE }
+    let(:config) {
+      double(
+        :config,
+        :args        => "doesn't matter",
+        :env         => {},
+        :upload_path => "arbitrary",
+        :remote?     => false,
+        :path        => nil,
+        :inline      => script,
+        :binary      => false,
+      )
+    }
+
+    it "does not raise an exception" do
+      vsp = VagrantPlugins::Shell::Provisioner.new(machine, config)
+
+      RSpec::Expectations.configuration.on_potential_false_positives = :nothing
+      # This test should be fine, since we are specifically looking for the
+      # string 'freeze' when RuntimeError is raised
+      expect {
+        vsp.provision
+      }.not_to raise_error(RuntimeError, /freeze/)
```

Comment:
@chrisroberts - question on this. I can't actually seem to get this to fail prior to my fix in unit tests (it does get fixed when I manually test this however). I even can show through `puts` statements that the script I've passed in is frozen with the `frozen?` method, but it doesn't seem to throw an exception and fail the test. Any idea why?